### ISSUE TITLE
CustomShaderMaterial does not support clone()

### ISSUE THREAD

**Issue by vincent-lecrubier-skydio** [FarazzShaikh/THREE-CustomShaderMaterial#19]
## Problem

Trying to `clone()` `CustomShaderMaterial` does not work currently and gives this error:

```
TypeError: Cannot destructure property 'baseMaterial' of 'undefined' as it is undefined.
```

## Cause

The reason is that the clone method is inherited from `THREE.Material` and looks like this 

```typescript
	clone() {
		return new this.constructor().copy( this );
	}
```

But the constructor looks like this (it destructures its first argument, which is not provided by clone).

```typescript
  constructor({
    baseMaterial,
    fragmentShader,
    vertexShader,
    uniforms,
    patchMap,
    cacheKey,
    ...opts
  }) {
    ...
  }
```

## Workaround

```typescript
/**
 * This class is a workaround to make CustomShaderMaterial support clone.
 */
class CustomShaderMaterial2 extends CustomShaderMaterial {
  parameters: iCSMParams;
  constructor(parameters: iCSMParams) {
    super(parameters);
    this.parameters = parameters;
  }
  clone() {
    return new this.constructor(this.parameters).copy(this);
  }
}
```

**Comment by FarazzShaikh** [comment]
Thanks for the report, will fix!

**Comment by Benjythebee** [comment]
This is an issue again in version 6.1.1 when using vanilla

**Comment by FarazzShaikh** [comment]
You are right. Will fix

**Comment by FarazzShaikh** [comment]
@Benjythebee Should be fixed in 6.2.0 please confirm.

**Comment by Benjythebee** [comment]
Looks fixed! Thanks @FarazzShaikh
